enhancements/insights/up-to-date-gathering.md

```diff
@@ -0,0 +1,250 @@
+6 days after we creted PR for 4.4 it is still not merged (now e2e tests are failing)
+We still need to merge it to 4.3.
+Since the beginning it is now 24 days.
+
+When CCX needs release faster: 
+- To identify affected customers (suggest Workaround), 
+- To find real feature usage, especially in in historical releases (because all that waiting on merge)
+- Fix IO bug
+
+
+### Goals
+
+As part of this enhancement we plan to do the following:
+* Release process change. On a fast-track release, publish Insights Operator fast-track changes during build to new quay tag for example 4.5-fast-track.
+The rest of release process will behave like today, release to 4.x.
+
+There will be two containers in Insights Operator Pod (*main* container with driver and *gatherer* container with Insights Operator).
+* Run the gathering logic in a separate container: a *gatherer* 
+* Control the lifecycle of the gatherer from *main* insights-operator container (different executable as part of the same image): this includes using newer versions
+of the image when available, as well as fallback mechanism to older version on failures.
+
+### Non-Goals
+
+
+## Proposal
+
+Separate The Insights Operator pod in two containers the Operator *main* and 
+Insighs Operator *gatherer*.
+
+The *main* will be similar to today's operator. It would like a proper operator 
+drive *gatherer*, but keep all the non-gathering logic, like uploading, status updates etc in it.
+It will be periodically checking for new versions in a 4.x-fast-track image on quay.io and eventually update the
+Insights Operator Gatherer container to latest 4.x-fast-track image url.
+It will not contain the Gathering methods and it will use Rest to reach *gatherer*, which will be 
+responsible for collection and saving data to storagePath (/var/lib/insights-operator) and this volume will be 
+shared among two containers.
+The *main* will instead of local Gather call remote Gather as Api, but will keep all the status handling and uploading
+in *main*.
+
+The Insights Operator *gatherer* will become just rest server on operators framework exposing Gather endpoint.
+Instead of returning all the data, the method will save the data using disk flusher to storagePath and return
+only status. (This was discussed as an improvement to save memory in IO for larger datasets, like logs).
+It would have same security role as today.
+The proposal will be implemented in three phases, described in [Implementation phases](#implementation-phases).
+
+### Integrating with CVO
+
+There is still to be described how we are going to keep upgraded Operator Deployment after the image url change.
+Cluster Version Operator is reverting Operator Deployment config even if I just add a new container.
+There are two possible ways. Create separate, non CVO managed Deployment for Gatherer and make Gatherer separate Pod,
+or find some way how to make CVO keep changes in second container image after the Gatherer image update. This is still
```

Comment:
Even if CVO didn't revert the changes periodically, the insights-operator managing a separate "gathering" deployment and its images sounds more compelling to me than it updating it's own deployment.

Reply:
You are right. Because of CVO reconcillation, we would need to create separate Pod (I guess ReplicaSet) dynamically and then drive the Image version in it.